frontend/src/scenes/surveys/surveyLogic.tsx

```diff
@@ -1258,15 +1265,23 @@ export const surveyLogic = kea<surveyLogicType>([
 
                     const data: number[] = questionResults.data
                     if (data.length === 11) {
-                        const promoters = data.slice(9, 11).reduce((a, b) => a + b, 0)
-                        const passives = data.slice(7, 9).reduce((a, b) => a + b, 0)
-                        const detractors = data.slice(0, 7).reduce((a, b) => a + b, 0)
-                        const npsScore = ((promoters - detractors) / (promoters + passives + detractors)) * 100
-                        return npsScore.toFixed(1)
+                        return calculateNpsScore(calculateNpsBreakdown(questionResults)).toFixed(1)
```

Comment:
we check if total == 0 and length == 1, outside of the method call, and now inside of `calculateNpsBreakdown`
so I think we should remove the one that is not needed.
`calculateNpsScore` returns 0 if `total === 0`, but above we return `No data available`
so this is also changing the behaviour and not only a refactoring

Reply:
`calculateNpsBreakdown` and `calculateNpsCore` are pure functions just doing those calculation - I don't think we should concern ourselves with the `surveyLogic` business rules, such as showing `No data available`

the part that we're refactoring is the actual calculation only. does that make sense?

Reply:
yes, it does, but we are doing the very same check twice now, and this issue is part of the refactoring.

Reply:
[did some changes here](https://github.com/PostHog/posthog/pull/29111/commits/55e04af2e25e2fbb7ecc8b974418187ac5394776)

now, to better mimic the behavior, instead of returning `0` for invalid data, which doesn't make much sense on second thought, I'm returning `null`. made the update to the logic/tests accordingly